server/http_integration_test.go

```diff
@@ -2585,3 +2586,61 @@ func TestCORS_Configuration(t *testing.T) {
 		})
 	}
 }
+
+func TestEndpoint_ResponseNilEvaluation(t *testing.T) {
+	client := newClient()
+
+	shutdown, hook := newCouper("testdata/integration/endpoint_eval/18_couper.hcl", test.New(t))
+	defer shutdown()
+
+	type testCase struct {
+		path   string
+		expVal bool
+	}
+
+	for _, tc := range []testCase{
+		{"/1stchild", true},
+		{"/2ndchild/no", true}, // TODO: write NilVal header vals?
+		{"/child-chain/no", true},
+		{"/list-idx", true},
+		{"/list-idx-splat", true},
+		{"/list-idx/no", true}, // TODO: handle nilVal / empty header
+		{"/list-idx-chain/no", true},
+		{"/list-idx-key-chain/no", true},
+		{"/root/no", true},
+		{"/tpl", true},
+		{"/for", true},
+	} {
+		t.Run(tc.path[1:], func(subT *testing.T) {
+			helper := test.New(subT)
+
+			req, err := http.NewRequest(http.MethodGet, "http://localhost:8080"+tc.path, nil)
+			helper.Must(err)
+
+			hook.Reset()
+			defer func() {
+				if subT.Failed() {
+					time.Sleep(time.Millisecond * 100)
+					for _, entry := range hook.AllEntries() {
+						s, _ := entry.String()
+						println(s)
+					}
+				}
+			}()
+
+			res, err := client.Do(req)
+			helper.Must(err)
+
+			if res.StatusCode != http.StatusOK {
+				subT.Errorf("Expected Status OK, got: %d", res.StatusCode)
+				return
+			}
+
+			val, ok := res.Header[http.CanonicalHeaderKey("X-Value")]
+			if !tc.expVal && ok {
```

Comment:
Shouldn't you also test the opposite case (expected but not present)?

Reply:
With https://github.com/avenga/couper/pull/225/commits/86d7b8d66f56159deed4a05e20812ae1c6daac16 it's possible now. Updated.